compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSTypeReferenceDeferredImpl.kt

```diff
@@ -19,10 +19,18 @@ package com.google.devtools.ksp.symbol.impl.kotlin
 
 import com.google.devtools.ksp.symbol.*
 import com.google.devtools.ksp.symbol.impl.KSObjectCache
-
-class KSTypeReferenceDeferredImpl private constructor(private val resolver: () -> KSType) : KSTypeReference {
-    companion object : KSObjectCache<() -> KSType, KSTypeReferenceDeferredImpl>() {
-        fun getCached(resolver: () -> KSType) = cache.getOrPut(resolver) { KSTypeReferenceDeferredImpl(resolver) }
+import com.google.devtools.ksp.symbol.impl.getInstanceForCurrentRound
+
+class KSTypeReferenceDeferredImpl private constructor(
+    private val resolver: () -> KSType,
+    override val parent: KSNode?
+) : KSTypeReference {
+    companion object : KSObjectCache<Pair<() -> KSType, KSDeclaration>, KSTypeReferenceDeferredImpl>() {
```

Comment:
we probably don't need the function type as key. they only depend on the location of the function.